### ISSUE TITLE
Configs: Interfaces and Types are not interchangeable

### ISSUE THREAD

**Issue by GauBen** [typescript-eslint/typescript-eslint#5343]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22preset+config+change%22+) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Description

https://github.com/typescript-eslint/typescript-eslint/blob/6fd476c32c4757cb9f4c442f0cd92875671eed30/packages/eslint-plugin/src/configs/recommended.ts#L15

The value should be `["error", {"allowSingleExtends": true}]` because interfaces and types are not interchangeable, especially in ambient type declarations.

Example with sveltekit:

```ts
declare namespace App {
  interface Session {
    thing: number;
	stuff: Record<string, string>;
  }

  interface Locals extends Session {}
  // `type Locals = Session` would not work

  // interface Platform {}
  // interface Stuff {}
}
```

**Comment by bradzacher** [comment]
If you're performing extensions on existing types, then they should really be done within `.d.ts`. The rule is built to handle this case.

If you're declaration merging randomly in `.ts` files, then the rule doesn't handle this.

Gonna need a better example as a small, isolated code fragment alone isn't really explanatory of your usecase. I am not familiar with sveltekit.

**Comment by bradzacher** [comment]
The lint rule cannot know for certain that you're declaring an interface for declaration merging or not. So it assumes you're not given that over 99% of the time you are not doing declaration merging.

However because there is this rare case - the fixer is disabled if you have an empty interface in a `.d.ts` file within a `declare namespace` block - we don't want to autofix this case and introduce broken code!
However, the rule still reports in case it is the majority case where you used the wrong construct.

You should use a disable comment here to signal this is an intentional empty interface for declaration merging.

We will not be changing the default config here.
